src/Mod/TechDraw/App/DimensionFormatter.cpp

```diff
@@ -321,6 +322,27 @@ QString DimensionFormatter::formatValueToSpec(const double value, const QString&
         // Second, try to cut also decimal dot and zeros, if there are just zeros after it
         formattedValue.replace(QRegularExpression(QStringLiteral("([0-9][0-9]*\\.[0-9]*[1-9])00*$")), QStringLiteral("\\1"));
         formattedValue.replace(QRegularExpression(QStringLiteral("([0-9][0-9]*)\\.0*$")), QStringLiteral("\\1"));
+    }
+    else if (formatSpecifier.contains(QRegularExpression(QStringLiteral("%.*[rR]")))) {
+        // round the value to the given precision
+        QString fs = formatSpecifier;
```

Comment:
Optional out of scope of this PR. Instead of copying the string (multiple places in the function), it should just be copied as a function parameter `QString&` -> `QString`